github_helper.py

```diff
@@ -0,0 +1,193 @@
+"""
+
+import dateutil.parser
+import requests
+import sys
+import urlparse
+
+GITHUB_API_ROOT = 'https://api.github.com'
+GITHUB_REPO_FMT_URL = GITHUB_API_ROOT + '/repos/{0}/{1}/'
+GITHUB_PULLS_ENDPOINT = 'pulls'
+
+BOT_NAME = 'merge-bot'
+GITHUB_SECRET = '../../github_auth/mergebot.secret'
+
+
+class GithubHelper(object):
+    """
+    GithubHelper provides a lightweight authenticated wrapper for the Github
+    API.
+    """
+
+    def __init__(self, org, repo):
+        self.repo_url = GITHUB_REPO_FMT_URL.format(org, repo)
+        with open(GITHUB_SECRET, 'r') as key_file:
+            self.bot_key = key_file.read().strip()
+        print_flush('Loaded key file.')
+
+    def fetch_prs(self):
+        """ Gets the PRs for the configured repository
+
+        Returns:
+            An iterable containing instances of GithubPR if successful.
+            Empty list otherwise.
+        """
+        prs = self.get(GITHUB_PULLS_ENDPOINT)
+        if prs is not None:
+            return [GithubPR(self, pr) for pr in prs]
+        return []
+
+    def get(self, endpoint):
+        """
+        get makes a GET request against a specified endpoint.
+
+        Args:
+            endpoint: URL to which to make the GET request. URL is relative
+            to https://api.github.com/repos/{self.org}/{self.repo}/
+        Returns:
+            JSON object retrieved from the endpoint.
+            None if there are none or an error is received.
+        """
+        url = urlparse.urljoin(self.repo_url, endpoint)
+        print_flush('Loading from Github at {}.'.format(url))
+        try:
+            resp = requests.get(url, auth=(BOT_NAME, self.bot_key))
+            if resp.status_code is 200:
+                return resp.json()
+            resp.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            print_flush("Non-200 HTTP Code Received:")
+            print_flush(e)
+        except requests.exceptions.Timeout as e:
+            print_flush('Request timed out:')
+            print_flush(e)
+        except requests.exceptions.RequestException as e:
+            print_flush('Catastrophic error in requests:')
+            print_flush(e)
+        return None
+
+    def post(self, content, endpoint):
+        """
+        post makes a POST request against a specified endpoint.
+
+        Args:
+            content: data to send in POST body.
+            endpoint: URL to which to make the POST request. URL is relative
+            to https://api.github.com/repos/{self.org}/{self.repo}/
+        Returns:
+            True if request completed successfully.
+            False otherwise.
+        """
+        payload = '{"body": "{}"}'.format(content)
+        url = urlparse.urljoin(self.repo_url, endpoint)
+        try:
+            resp = requests.post(url, data=payload, auth=(BOT_NAME,
+                                                          self.bot_key))
+            if resp.status_code // 100 is 2:
```

Comment:
Why the difference between here and above? For get we want exactly 200, but here we want any code in the 200 range?